src/formatDate.js

```diff
@@ -8,7 +9,41 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separador = fromFormat[3];
+  const separadorTo = toFormat[3];
+
+  const newDate = date.split(separador);
+
+  const dateObj = {};
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    dateObj[fromFormat[i]] = newDate[i];
+  }
+
+  const newDateParts = [];
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    const formatKey = toFormat[i];
+
+    if (formatKey === 'YYYY' && dateObj['YY']) {
+      // Converter YY para YYYY
+      const year = parseInt(dateObj['YY'], 10);
+
+      dateObj['YYYY'] = year < 30 ? '20' + dateObj['YY'] : '19' + dateObj['YY'];
+    }
+
+    if (formatKey === 'YY' && dateObj['YYYY']) {
+      // Converter YYYY para YY
+      dateObj['YY'] = dateObj['YYYY'].slice(-2);
```

Comment:
The conversion from 'YYYY' to 'YY' is correct, but you might consider checking if `dateObj['YYYY']` is not `undefined` before attempting to slice it to prevent potential errors if the input is not as expected.